modules/services/dunst.nix

```diff
@@ -10,29 +16,54 @@ with lib;
       enable = mkEnableOption "the dunst notification daemon";
 
       settings = mkOption {
-        type = types.attrs;
+        type = types.attrsOf types.attrs;
         default = {};
         description = "Configuration written to ~/.config/dunstrc";
+        example = literalExample ''
+          {
+            global = {
+              geometry = "300x5-30+50";
+              transparency = 10;
+              frame_color = "#eceff1";
+              font = "Droid Sans 9";
+              browser = "''${config.firefox.package}/bin/firefox -new-tab";
+            };
+
+            urgency_normal = {
+              background = "#37474f";
+              foreground = "#eceff1";
+              timeout = 10;
+            };
+          };
+        '';
       };
     };
   };
 
-  config = mkIf config.services.dunst.enable {
-    home.file.".local/share/dbus-1/services/org.knopwob.dunst.service".source =
-      "${pkgs.dunst}/share/dbus-1/services/org.knopwob.dunst.service";
+  config = mkIf cfg.enable (
+    mkMerge [
+      {
+        home.file.".local/share/dbus-1/services/org.knopwob.dunst.service".source =
+          "${pkgs.dunst}/share/dbus-1/services/org.knopwob.dunst.service";
 
-    systemd.user.services.dunst = {
-      Unit = {
-        Description = "Dunst notification daemon";
-        After = [ "graphical-session-pre.target" ];
-        PartOf = [ "graphical-session.target" ];
-      };
+        systemd.user.services.dunst = {
+          Unit = {
+            Description = "Dunst notification daemon";
+            After = [ "graphical-session-pre.target" ];
+            PartOf = [ "graphical-session.target" ];
+          };
 
-      Service = {
-        Type = "dbus";
-        BusName = "org.freedesktop.Notifications";
-        ExecStart = "${pkgs.dunst}/bin/dunst";
-      };
-    };
-  };
+          Service = {
+            Type = "dbus";
+            BusName = "org.freedesktop.Notifications";
+            ExecStart = "${pkgs.dunst}/bin/dunst";
+          };
+        };
+      }
+
+      (mkIf (cfg.settings != {}) {
+        home.file.".config/dunst/dunstrc".text = generators.toINI {} cfg.settings;
```

Comment:
I think the default `toINI` is insufficient here since it won't put quotes around strings, and that is needed for [`#rrggbb` style colors](https://github.com/dunst-project/dunst/blob/4d59e3a2e258d6ae6dd7ce110f60021a2b8b38ef/dunstrc#L246-L247).

Perhaps add

```
  toDunstIni = generators.toINI {
    mkKeyValue = key: value:
      let
        value' =
          if isBool value then (if value then "yes" else "no")
          else if isString value then "\"${value}\""
          else toString value;
      in
        "${key}=${value'}";
  };
```

to the `let` above and use `toDunstIni cfg.settings` here? This also writes out boolean values correctly.